python/lsst/ts/ATMCSSimulator/mcs_csc.py

```diff
@@ -472,35 +462,34 @@ def m3_port_rot(self, t):
             * rotator axis: the instrument rotator at this port,
               as an Axis enum value, or None if the port has no rotator.
         """
-        if not self.m3_in_position(t):
+        if not self.m3_in_position(tai):
             return (None, None)
-        cmd_pos = self.actuators[Axis.M3].cmd.p0
+        cmd_pos = self.actuators[Axis.M3].target.position
         for exit_port, (ind, m3state, rot_axis) in self._port_info_dict.items():
             if self.m3_port_pos[ind] == cmd_pos:
                 return (exit_port, rot_axis)
         return (None, None)
 
-    def m3_in_position(self, t):
+    def m3_in_position(self, tai):
         """Is the M3 actuator in position?
 
         Parameters
         ----------
-        t : `float`
+        tai : `float`
             Current time, TAI unix seconds.
         """
         m3actuator = self.actuators[Axis.M3]
-        if m3actuator.kind(t) != path.Kind.Stopped:
+        if m3actuator.kind(tai) != m3actuator.Kind.Stopped:
             return False
-        m3cmd_pos = m3actuator.cmd.p0
-        m3curr_pos, m3curr_vel = m3actuator.curr[-1].pva(t)[0:2]
-        return abs(m3cmd_pos - m3curr_pos) < self.m3tolerance
+        m3cmd_pos = m3actuator.target.position
+        m3curr = m3actuator.path[-1].at(tai)
```

Comment:
Since you have been very careful with variable naming making the code nicely readable, let me suggest to be more explicit with suffix `curr` in `m3curr`. Does it stand for `current`?
The same applies to the line above with `m3cmd_pos`. I prefer to have longer variable names always, but that is personal taste.